hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -195,14 +197,98 @@ def _exceptions_from_docstring(doc: str) -> Tuple[Type[Exception], ...]:
 # on analysis of type-annotated code to detect arguments which almost always
 # take values of a particular type.
 _GUESS_STRATEGIES_BY_NAME = (
+    (st.integers(0, 32), ["ndims"]),
+    (st.booleans(), ["keepdims"]),
     (st.text(), ["name", "filename", "fname"]),
     (st.floats(), ["real", "imag"]),
     (st.functions(), ["function", "func", "f"]),
+    (st.functions(returns=st.booleans(), pure=True), ["pred", "predicate"]),
     (st.iterables(st.integers()) | st.iterables(st.text()), ["iterable"]),
 )
 
 
-def _strategy_for(param: inspect.Parameter) -> Union[st.SearchStrategy, InferType]:
+def _type_from_doc_fragment(token: str) -> Optional[type]:
+    # Special cases for "integer" and for numpy array-like and dtype
+    if token == "integer":
+        return int
+    if "numpy" in sys.modules:
+        if re.fullmatch(r"[Aa]rray[-_ ]?like", token):
+            return sys.modules["numpy"].ndarray  # type: ignore
+        elif token == "dtype":
+            return sys.modules["numpy"].dtype  # type: ignore
+    # Natural-language syntax, e.g. "sequence of integers"
+    coll_match = re.fullmatch(r"(\w+) of (\w+)", token)
+    if coll_match is not None:
+        coll_token, elem_token = coll_match.groups()
```

Comment:
It might be worthwhile normalizing these tokens (e.g. by lower casing)

Reply:
That would break parsing for tokens like `sequence of Exceptions`.